packages/teleport/src/Support/ScheduleUpgrades/ScheduleUpgrades.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { ButtonPrimary, ButtonSecondary } from 'design/Button';
+import Dialog, {
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from 'design/Dialog';
+import Select, { Option } from 'shared/components/Select';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import { UpgradeWindowStart } from 'teleport/services/upgradeWindow/upgradeWindow';
+import Text from 'design/Text';
+import Alert from 'design/Alert';
+
+const availableUpgradeWindowStarts: UpgradeWindowStart[] = [
+  '08:00:00',
+  '16:00:00',
+  '23:00:00',
+];
```

Comment:
something about this was bugging me and i just realized it! it was weird to be defining these consts twice. i think this is what you wanted, we can derive the type from the array:

```ts
type UpgradeWindowStart = typeof availableUpgradeWindowStarts[number];
// you get:  "08:00:00" | "16:00:00" | "23:00:00"
```

so i think these definitions (array and type) actually belong in the service file?

Reply:
Awesome, this is what I wanted!
Btw I had to add `as const` after `availableUpgradeWindowStarts`, otherwise I get `type UpgradeWindowStart = string`:

```js
const availableUpgradeWindowStarts = ['08:00:00', '16:00:00', '23:00:00'];
export type UpgradeWindowStart = typeof availableUpgradeWindowStarts[number];
// type UpgradeWindowStart = string

const availableUpgradeWindowStarts = ['08:00:00', '16:00:00', '23:00:00'] as const;
export type UpgradeWindowStart = typeof availableUpgradeWindowStarts[number];
//type UpgradeWindowStart = "08:00:00" | "16:00:00" | "23:00:00"

```